config/migrations/20200907140400-treatments-to-documents.sparql

```diff
@@ -0,0 +1,27 @@
+PREFIX besluitvorming: <http://data.vlaanderen.be/ns/besluitvorming#>
+PREFIX besluit:  <http://data.vlaanderen.be/ns/besluit#>
+PREFIX dossier: <https://data.vlaanderen.be/ns/dossier#>
+PREFIX pav: <http://purl.org/pav/>
+
+DELETE
+{
+GRAPH ?g {
+    ?treatment besluitvorming:genereertVerslag ?verslag.
+}
+}
+INSERT
+ {
+ GRAPH ?g {
+    ?treatment besluitvorming:genereertVerslag ?rightdoc.
+ }
+ }
+WHERE
+{
+GRAPH ?g {
+    ?treatment a besluit:BehandelingVanAgendapunt.
+    ?treatment besluitvorming:genereertVerslag ?verslag.
+    ?verslag a dossier:Serie.
+    ?verslag dossier:collectie.bestaatUit ?rightdoc.
+    FILTER NOT EXISTS {?rightdoc ^pav:previousVersion ?otherDoc.}
```

Comment:
Lees je eens mee de beschrijving van [pav:previousVersion](http://pav-ontology.github.io/pav/index.html#http://purl.org/pav/previousVersion)? Volgens mij is het hoedje er te veel aan ... Verwarrend wel, wanneer syntax en semantiek samen komen ;-)

Reply:
De query is wel verwarrend met die filter.
Ik vermoed wat je hier wilt bereiken is de laatste versie ophalen van de document container.
Dus dat vertaald zich dan in het dossier#Stuk waar er geen volgende versie van bestaat?
Een stuk zonder volgende versie is ofwel de enigste versie, of de laatste versie in een reeks

Dus dan zou je dit kunnen doen :
`FILTER NOT EXISTS { ?otherDoc pav:previousVersion ?rightdoc. }`
of wat jij hebt gedaan, met het hoedje
`FILTER NOT EXISTS {?rightdoc ^pav:previousVersion ?otherDoc.}`

@MikiDi Klopt dit ?

Reply:
Helemaal gelijk :ok_hand: ... :brain: had gedetecteerd dat het een instinker was, en ben er toch in getrapt :facepalm: